core/modules/system/system.install

```diff
@@ -14,7 +14,129 @@ function system_requirements($phase) {
+      $output = preg_replace('/[\s]+/mu', ' ', $output);
+      // Convert any special characters to underscore.
+      $output = preg_replace('/[^a-zA-Z0-9_\s]/', '_', $output);
+
+      $result[$line] = explode(" ", $output);
+    }
+
+    // Grab the 1st line of the output (which is the heading of the various
+    // columns), then remove that from the rest of the output.
+    $header = $result[0];
+    unset($result[0]);
+    // Grab each RAM metric (per column).
+    $mem_metrics = array();
+    foreach ($result as $result_line => $data) {
+      // The first column contains the type of memory ("mem" or "swap").
+      $prefix = $data[0];
+      array_splice($data, 0, 1);
+      foreach ($data as $column => $metric) {
+        $mem_metrics[$prefix . $header[$column]] = $metric;
+      }
+    }
+
+    $mem_usage = round(($mem_metrics['mem_total'] - $mem_metrics['mem_available']) / $mem_metrics['mem_total'] * 100);
+
+    // Disk metrics. Should work the same across operating systems.
+    $disk_total = disk_total_space('.');
+    $disk_free = disk_free_space('.');
+    $disk_used = $disk_total - $disk_free;
+    $disk_usage = round($disk_used / $disk_total * 100);
+
+    // Sizes of the various files directories.
+    $directories = array(
+      'temp' => file_directory_temp(),
+      'public' => backdrop_realpath('public://'),
+      'private' => backdrop_realpath('private://'),
+    );
+    $directory_sizes = array();
+    $total_dir_size = 0;
+    foreach ($directories as $directory => $path) {
+      if (!empty($path)) {
+        $dir_size = _system_get_dir_size($path);
+        $directory_sizes[$directory] = $t('@dir: @size', array('@dir' => $directory, '@size' => format_size($dir_size)));
+        $total_dir_size += $dir_size;
+      }
+    }
+    $dir_size_breakdown = theme('item_list', array('items' => $directory_sizes));
+
+    // Database metrics.
+    $rows = db_query('SHOW TABLE STATUS')->fetchAll();
+    $db_size = 0;
+    $db_cache_size = 0;
+    foreach ($rows as $row) {
+      if (strpos($row->Name, 'cache') === 0) {
+        $db_cache_size += $row->Data_length + $row->Index_length;
+      }
+      $db_size += $row->Data_length + $row->Index_length;
+    }
+
+    // PHP memory usage.
+    $php_mem_usage = format_size(memory_get_usage());
+    $php_mem_peak_usage = format_size(memory_get_peak_usage());
+    $php_mem_true_usage = format_size(memory_get_peak_usage(TRUE));
+
+    // CPU load - not implemented on Windows platforms.
+    if (stristr(PHP_OS, 'win') === FALSE) {
+      $cpu_load = sys_getloadavg();
+      // This is not a percentage. For details, see:
+      // - https://stackoverflow.com/questions/9719155/understading-sys-getloadavg
+      // - https://www.php.net/manual/function.sys-getloadavg.php
+      $cpu_metric = $t('CPU load (over 1, 5 and 15 minutes): @load', array('@load' => implode(' / ', $cpu_load)));
+    }
+    else {
+      $cpu_metric = $t('CPU load cannot be estimated on Windows platforms');
```

Comment:
Backdrop hosting on Windows? Really?

Reply:
Apparently some people do that 🤷🏼 ...Amazon offers Windows servers on the cloud: https://aws.amazon.com/windows

...and our telemetry shows that there are few sites running on Windows (although I'm assuming that this is most likely on local, before pushing changes to a linux seever)
![image](https://user-images.githubusercontent.com/2423362/201542278-02879bbb-3eaf-42de-8c24-2d68ae671783.png)

Anyway, this was some "defensive coding", to account the display of this when developing on Windows local, running WAMPs etc.

Reply:
> ...and our telemetry shows that there are few sites running on Windows...

One percent. :smirk_cat: And, yes, I agree that these are very likely local dev instances.